core/src/test/java/io/questdb/test/griffin/engine/functions/bind/BindVariableServiceImplTest.java

```diff
@@ -1026,4 +1051,24 @@ public void testTimestampVarSetToInt() throws Exception {
             Assert.assertEquals(Numbers.LONG_NULL, d);
         });
     }
+
+    private static void testSetIntoUndefined0(int index, RunnableWithSqlException test, int expectedType) throws SqlException {
+        Function funA = bindVariableService.getFunction(index);
+        Assert.assertNull(funA);
+        Function funB = bindVariableService.getOrDefineFunction(index);
+        funA = bindVariableService.getFunction(index);
+        Assert.assertSame(funB, funA);
+        Assert.assertTrue(funB.isUndefined());
+
+        test.run();
+
+        funA = bindVariableService.getFunction(index);
+        funB = bindVariableService.getOrDefineFunction(index);
+        Assert.assertSame(funB, funA);
+        Assert.assertEquals(expectedType, funB.getType());
+    }
+
+    private static interface RunnableWithSqlException {
```

Comment:
nit: `static` modifier is redundant
```suggestion
    private interface RunnableWithSqlException {
```